addons/project/wizard/project_task_convert_subtask.py

```diff
@@ -0,0 +1,33 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class ProjectTaskConvertSubtask(models.TransientModel):
+    _name = 'project.task.convert.subtask.wizard'
+    _description = 'Project Task Convert to Subtask Wizard'
+
+    project_id = fields.Many2one('project.project', compute='_compute_project_id', help="Project of parent task",
+        readonly=False)
+    task_ids = fields.Many2many('project.task', string='Sub Tasks')
+    parent_id = fields.Many2one('project.task', string='Parent Task', readonly=False, compute="_compute_parent_id",
+        domain="[('id', 'not in', task_ids), ('project_id', '!=', False)]", store=True)
```

Comment:
So you could select as parent a child of the selected tasks ?

Reply:
Yes, you can select a parent task that is a child of the selected tasks, but it will be skipped [here](https://github.com/odoo/odoo/pull/144179/files#diff-c790cfa8b1b0b6bcc54ff491d298861eaca514e62858b65dc773b22a7b97104cR33)